Module-generation/Jenkinsfile

```diff
@@ -0,0 +1,76 @@
+
+def archive() {
+    def changes=false
+    def equals = sh( 
+        
+    script: '''
+     #!/bin/bash 
+
+     DIFF=$(diff -r modules/ artifacts/modules/ | tail -1)
+     if [ "$DIFF" != "" ]; then
+     echo "Not same"
+     else
+     echo "Same"
+     fi
+
+     ''', returnStdout: true).trim()
+
+    if(equals == "Same"){
+         println("Same as old artifact.")
+    }else{
+         println("Some files are updated.") 
+         changes=true        
+    }
+
+    return changes
+}
```

Comment:
You do not need all this.
The exit status on running `diff` is sufficient to tell whether things are different or not.

From the manpage: "Exit status is 0 if inputs are the same, 1 if different, 2 if trouble."